docs_new/features/core/generator.md

```diff
@@ -0,0 +1,161 @@
+  - Only supported for `SteerableModel` instances.
+
+ The `output_type` and `processor` parameters are mutually exclusive. Provide at most one.
+
+## Returns
+
+The constructor returns a `generator` object that contains the methods below. If you provide an async model, you will get an async generator.
+
+### `__call__(prompt, **inference_kwargs)`
+
+Generate text for the given prompt with the model given to the `Generator`. The response will follow the structure provided through the output type.
+
+- **prompt**: The input text prompt.
+- **inference_kwargs**: Additional keyword arguments passed to the model's text generation function.
+- **Returns**: The generated text conforming to the specified output type.
+
+### `stream(prompt, **inference_kwargs)`
+
+Generate text generation for the given prompt with the model given to the `Generator`. The response will follow the structure provided through the output type.
+
+- **prompt**: The input text prompt.
+- **inference_kwargs**: Additional keyword arguments passed to the model's text generation function.
+- **Returns**: An iterator (or async iterator for async models) that yields text chunks as they're generated.
+
+#### Text generation return type
+
+The `__call__` and `stream` methods described above always return a raw strings (or a list of strings in case of batch generation), even when using constrained generation. You must cast the string into the desired format. For instance, if you provided a Pydantic model as output type, you should use the `model_validate_json` of the model to parse the generation response as a Pydantic model instance.
+
+## Usage
+
+### Basic Usage
+
+```python
+from outlines import Generator, from_transformers
+import transformers
+
+# Create a model
+model = from_transformers(
+    transformers.AutoModelForCausalLM.from_pretrained("microsoft/Phi-3-mini-4k-instruct"),
+    transformers.AutoTokenizer.from_pretrained("microsoft/Phi-3-mini-4k-instruct"),
+)
+
+# Create a generator for unconstrained text and call it to generate text
+generator = Generator(model)
+response = generator("Tell me about machine learning.", max_new_tokens=20)
+
+print(result) # "Machine learning is a subset of artificial intelligence (AI) that provides..."
+```
+
+### Generating Structured Output
+
+```python
+from typing import List
+from pydantic import BaseModel
+from outlines import Generator, from_llamacpp
+from llama_cpp import Llama
+
+# Create a model
+model = from_llamacpp(
+    Llama.from_pretrained(
+        repo_id="M4-ai/TinyMistral-248M-v2-Instruct-GGUF",
+        filename="TinyMistral-248M-v2-Instruct.Q4_K_M.gguf",
+    )
+)
+
+# Define the output structure
+class Character(BaseModel):
+    name: str
+    age: int
+    skills: List[str]
+
+# Create a generator for the output type defined and call it to generate text
+generator = Generator(model, Character)
+response = generator("Create a character for my game", max_tokens=100)
```

Comment:
The prompt here is not correct because we are using an instruction-tuned model without user and assistant tags. The model will generate a character, but anyone using this code will find severely degraded performance in more flexible contexts. 

Brief overview [here](https://dottxt-ai.github.io/outlines/main/reference/chat_templating/).

This is more of a general issue (feature?) of how Outlines works that we need to find a way to consistently address. The transformer tokenizer approach is the most ergonomic, and I'm wondering if we should just offer that everywhere even for different inference backends.